homework/list.js

```diff
@@ -1,79 +1,108 @@
 'use strict';
 
-(function() {
+(function(exports) {
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
-
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
+  /**
+   * Creates an instance of ListManager.
+   *
+   * @constructor
+   * @this {ListManager}
+   */
+  var ListManager = function() {
+    this._listNoteContent = [];
+    this._wrapper = document.querySelector('#note-list-wrapper');
   }
 
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    };
-  }
+  ListManager.prototype = {
 
-  function preloadFirstNote() {
-    if (_listNoteContent.length !== 0) {
-      var content = _listNoteContent[0];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    }
-  }
+    /**
+     * Binding the 'click' event to the '#note-list-wrapper' node and
+     * fetch the data from server then draw it.
+     * This method should be called only once to initialize the ListManager.
+     *
+     * @this {ListManager}
+     */
+    start() {
+      this.fetchList().then((function(data){
+        this.updateList(data);
+        this.drawList();
+        this.preloadFirstNote();
+      }).bind(this)).catch((function() {
 
-  function updateList(list) {
-    _listNoteContent = list;
-  }
+      }).bind(this));
+      window.addEventListener('click', (function(event) {
+        this.onNoteOpen(event);
+      }).bind(this));
+    },
 
-  function drawList() {
-    var list = _listNoteContent;
-    var ul = document.createElement('ul');
-    ul.id = 'note-title-list';
-    var buff = document.createDocumentFragment();
-    list.forEach(function(note, i) {
-      var li = document.createElement('li');
-      li.dataset.noteId = i;
-      li.classList.add('note-title');
-      li.textContent = note.title;
-      // Note: buff is captured, so we now have a
-      // little closure naturally.
-      buff.appendChild(li);
-    });
-    ul.appendChild(buff);
-    _wrapper.appendChild(ul);
-  }
+    /**
+     * Dispatch a CustomEvent 'note-open' with note detail to window for others to use.
+     *
+     * @param {CustomEvent} event - Which event.target has 'note-title' in its classList.
+     * @this {ListManager}
+     */
+    onNoteOpen(event) {
```

Comment:
It's a convention to prefix the `on` to the handler method with corresponding event. Therefore this method should be renamed as `onNoteClick`.